### PR TITLE
improve(gas-rebate): Gas Rebate 12 + improved error handling

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

<details>
  <summary>Expand script receipt</summary>

```
nicholaspai@Nicholass-MacBook-Pro-744 ➜  protocol git:(master) ✗ node ./packages/affiliates/gas-rebate/VoterGasRebate.js  --rebateNumber 12 --start-block 12775311 --network mainnet_mnemonic


*=======================================*
*                                       *
* 🐣 Setup 🐣                           *
* - Fetching block number for timestamp *
* - Fetching historical gas px data     *
* - Fetching UMA-ETH exchange rates     *
*                                       *
*=======================================*
- ✅ Success, running main script now
Using DVM @ 0x8B1631ab830d11531aE83725fDa4D86012eCCd77


*=======================================*
*                                       *
* 🐲⛽️ UMA Gas Rebater 🐲 ⛽️            *
*                                       *
*=======================================*
- Calculating gas rebates from block 12775311 until 13031692
- dailyAvgGasPrices: [
    {
        "timestamp": 1625529600,
        "avgGwei": "54.871815247"
    },
    {
        "timestamp": 1625616000,
        "avgGwei": "40.369381391"
    },
    {
        "timestamp": 1625702400,
        "avgGwei": "37.478388049"
    },
    {
        "timestamp": 1625788800,
        "avgGwei": "30.266605013"
    },
    {
        "timestamp": 1625875200,
        "avgGwei": "18.988461455"
    },
    {
        "timestamp": 1625961600,
        "avgGwei": "15.796724325"
    },
    {
        "timestamp": 1626048000,
        "avgGwei": "30.407815045"
    },
    {
        "timestamp": 1626134400,
        "avgGwei": "32.450026038"
    },
    {
        "timestamp": 1626220800,
        "avgGwei": "45.974504881"
    },
    {
        "timestamp": 1626307200,
        "avgGwei": "45.196906108"
    },
    {
        "timestamp": 1626393600,
        "avgGwei": "42.029589044"
    },
    {
        "timestamp": 1626480000,
        "avgGwei": "29.976286417"
    },
    {
        "timestamp": 1626566400,
        "avgGwei": "30.497194578"
    },
    {
        "timestamp": 1626652800,
        "avgGwei": "30.440382"
    },
    {
        "timestamp": 1626739200,
        "avgGwei": "35.422941438"
    },
    {
        "timestamp": 1626825600,
        "avgGwei": "27.173423658"
    },
    {
        "timestamp": 1626912000,
        "avgGwei": "33.177581738"
    },
    {
        "timestamp": 1626998400,
        "avgGwei": "35.451146095"
    },
    {
        "timestamp": 1627084800,
        "avgGwei": "24.896279188"
    },
    {
        "timestamp": 1627171200,
        "avgGwei": "23.442307403"
    },
    {
        "timestamp": 1627257600,
        "avgGwei": "42.016473026"
    },
    {
        "timestamp": 1627344000,
        "avgGwei": "40.656771727"
    },
    {
        "timestamp": 1627430400,
        "avgGwei": "46.698873266"
    },
    {
        "timestamp": 1627516800,
        "avgGwei": "34.329937499"
    },
    {
        "timestamp": 1627603200,
        "avgGwei": "39.490647512"
    },
    {
        "timestamp": 1627689600,
        "avgGwei": "44.287844806"
    },
    {
        "timestamp": 1627776000,
        "avgGwei": "39.568132114"
    },
    {
        "timestamp": 1627862400,
        "avgGwei": "44.396980231"
    },
    {
        "timestamp": 1627948800,
        "avgGwei": "48.965615995"
    },
    {
        "timestamp": 1628035200,
        "avgGwei": "45.772397433"
    },
    {
        "timestamp": 1628121600,
        "avgGwei": "64.498157332"
    },
    {
        "timestamp": 1628208000,
        "avgGwei": "54.09541195"
    },
    {
        "timestamp": 1628294400,
        "avgGwei": "53.823017552"
    },
    {
        "timestamp": 1628380800,
        "avgGwei": "50.74058315"
    },
    {
        "timestamp": 1628467200,
        "avgGwei": "62.245603314"
    },
    {
        "timestamp": 1628553600,
        "avgGwei": "65.215176708"
    },
    {
        "timestamp": 1628640000,
        "avgGwei": "67.30278672"
    },
    {
        "timestamp": 1628726400,
        "avgGwei": "58.916336779"
    },
    {
        "timestamp": 1628812800,
        "avgGwei": "62.916929886"
    },
    {
        "timestamp": 1628899200,
        "avgGwei": "53.617974436"
    },
    {
        "timestamp": 1628985600,
        "avgGwei": "62.553842764"
    }
]
- dailyAvgUmaEthPrices: [
    {
        "timestamp": 1625529600,
        "avgPx": "0.004249532960570039"
    },
    {
        "timestamp": 1625616000,
        "avgPx": "0.004291248041432338"
    },
    {
        "timestamp": 1625702400,
        "avgPx": "0.004246967532942191"
    },
    {
        "timestamp": 1625788800,
        "avgPx": "0.004327140122371156"
    },
    {
        "timestamp": 1625875200,
        "avgPx": "0.004416839534732457"
    },
    {
        "timestamp": 1625961600,
        "avgPx": "0.004303562855013598"
    },
    {
        "timestamp": 1626048000,
        "avgPx": "0.004281986304028872"
    },
    {
        "timestamp": 1626134400,
        "avgPx": "0.004350631212317500"
    },
    {
        "timestamp": 1626220800,
        "avgPx": "0.004375617577444560"
    },
    {
        "timestamp": 1626307200,
        "avgPx": "0.004340341633596533"
    },
    {
        "timestamp": 1626393600,
        "avgPx": "0.004284604394589153"
    },
    {
        "timestamp": 1626480000,
        "avgPx": "0.004246396413548258"
    },
    {
        "timestamp": 1626566400,
        "avgPx": "0.004519323312322176"
    },
    {
        "timestamp": 1626652800,
        "avgPx": "0.004322985685117200"
    },
    {
        "timestamp": 1626739200,
        "avgPx": "0.004398749967888575"
    },
    {
        "timestamp": 1626825600,
        "avgPx": "0.004319161346439285"
    },
    {
        "timestamp": 1626912000,
        "avgPx": "0.004250204747271410"
    },
    {
        "timestamp": 1626998400,
        "avgPx": "0.004041884495629951"
    },
    {
        "timestamp": 1627084800,
        "avgPx": "0.004215915979319269"
    },
    {
        "timestamp": 1627171200,
        "avgPx": "0.004006230296037555"
    },
    {
        "timestamp": 1627257600,
        "avgPx": "0.003899932798610759"
    },
    {
        "timestamp": 1627344000,
        "avgPx": "0.003845334328851640"
    },
    {
        "timestamp": 1627430400,
        "avgPx": "0.003856099829152790"
    },
    {
        "timestamp": 1627516800,
        "avgPx": "0.003731770255217574"
    },
    {
        "timestamp": 1627603200,
        "avgPx": "0.003717146557466030"
    },
    {
        "timestamp": 1627689600,
        "avgPx": "0.003765782174723034"
    },
    {
        "timestamp": 1627776000,
        "avgPx": "0.003591520455716020"
    },
    {
        "timestamp": 1627862400,
        "avgPx": "0.003594742462834510"
    },
    {
        "timestamp": 1627948800,
        "avgPx": "0.003450969757606905"
    },
    {
        "timestamp": 1628035200,
        "avgPx": "0.003331069949858441"
    },
    {
        "timestamp": 1628121600,
        "avgPx": "0.003376933402099709"
    },
    {
        "timestamp": 1628208000,
        "avgPx": "0.003372822214137384"
    },
    {
        "timestamp": 1628294400,
        "avgPx": "0.003360684614375253"
    },
    {
        "timestamp": 1628380800,
        "avgPx": "0.003437530788104591"
    },
    {
        "timestamp": 1628467200,
        "avgPx": "0.003283012476586486"
    },
    {
        "timestamp": 1628553600,
        "avgPx": "0.003125671594199269"
    },
    {
        "timestamp": 1628640000,
        "avgPx": "0.003634239694176830"
    },
    {
        "timestamp": 1628726400,
        "avgPx": "0.003515295755124578"
    },
    {
        "timestamp": 1628812800,
        "avgPx": "0.003565854299924434"
    },
    {
        "timestamp": 1628899200,
        "avgPx": "0.003516391092247526"
    },
    {
        "timestamp": 1628985600,
        "avgPx": "0.003578115203092821"
    }
]


*=======================================*
*                                       *
* 🌏 Fetching Blockchain Data 🌎        *
*                                       *
*=======================================*
Reveal Events [████████████████████████████████████████] 99% | ⏳ ETA: 1s | events parsed: 765/766
Claim Events [████████████████████████████████████████] 100% | ⏳ ETA: 0s | events parsed: 669/669

*=======================================*
*                                       *
* ✅ Results                           *
*                                       *
*=======================================*
🗄 257 Reveal Transactions successfully written to /Users/nicholaspai/UMA/protocol/packages/affiliates/gas-rebate/debug/Reveals_12.json
㊗️ Reveal Totals: {
  totalGasUsed: 89657745,
  totalEthSpent: 3.929223213430271,
  totalUmaRepaid: 1040.417802176576
}
🗄 90 Claim Transactions successfully written to /Users/nicholaspai/UMA/protocol/packages/affiliates/gas-rebate/debug/Claims_12.json
㊗️ Claim Totals: {
  totalGasUsed: 17818535,
  totalEthSpent: 0.7729247771484412,
  totalUmaRepaid: 209.07332587013656
}
```

</details>

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [X]  Untested

### CODE REVIEW

File: packages/affiliates/gas-rebate/VoterGasRebate.js [hunk 1]
```diff
@@ -101,70 +101,74 @@ async function parseRevealEvents({ committedVotes, revealedVotes, priceData, reb
   }
 
   for (let i = 0; i < revealedVotes.length; i++) {
-    const reveal = revealedVotes[i];
```
Comment (nicholaspai) [inline packages/affiliates/gas-rebate/VoterGasRebate.js:104]: Nothing changed in this code block but added a try-catch around it.

File: packages/affiliates/gas-rebate/VoterGasRebate.js [hunk 2]
```diff
@@ -558,6 +566,9 @@ async function getHistoricalGasPrice(startBlock, endBlock) {
 
     let data = (await response.json()).result;
 
+    if (!Array.isArray(data))
+      throw new Error("Unexpected ETH gas price historical data, did you set ETHERSCAN_API_KEY environment variable?");
```
Comment (daywiss) [inline packages/affiliates/gas-rebate/VoterGasRebate.js:570]: do you just want to assert for `ETHERSCAN_API_KEY` before running script, or is it optional?

**Review by daywiss — COMMENTED** [review]
just want to make sure that the script doesnt break if any of these blocks error, ie that the future sections dont depend on calculations from the previous sections. Maybe its not such a big deal, but could you comment on that? If thats all good it seems fine

File: packages/affiliates/gas-rebate/VoterGasRebate.js (hunk 2 above)
Comment (nicholaspai) [reply packages/affiliates/gas-rebate/VoterGasRebate.js:570]: oh good call! its definitely required

Comment (nicholaspai) [reply packages/affiliates/gas-rebate/VoterGasRebate.js:570]: I realize that I actually do this but its possible to set the incorrect api key, so i'll edit this error to just explain the data is ill formatted

**Review by daywiss — APPROVED** [review]
i just realized the try catches are all in loops, so it seems like you just miss an iteration of the loop if one fails. which seems fine.

**Review by claytonroche — APPROVED** [review]

**Pull request merged**